l1-contracts/test/Inbox.t.sol

```diff
@@ -0,0 +1,241 @@
+    DataStructures.L1ToL2Msg memory message = _fakeMessage();
+    inbox.sendL2Message{value: message.fee}(
+      message.recipient, message.deadline, message.content, message.secretHash
+    );
+    skip(1000); // block.timestamp now +1000 ms.
+    vm.expectRevert(Inbox.Inbox__NotPastDeadline.selector);
+    inbox.cancelL2Message(message, address(0x1));
+  }
+
+  function testRevertIfCancellingNonExistentMessage() public {
+    DataStructures.L1ToL2Msg memory message = _fakeMessage();
+    bytes32 entryKey = _helper_computeEntryKey(message);
+    vm.expectRevert(
+      abi.encodeWithSelector(MessageBox.MessageBox__NothingToConsume.selector, entryKey)
+    );
+    inbox.cancelL2Message(message, address(0x1));
+  }
+
+  function testCancelMessage() public {
+    DataStructures.L1ToL2Msg memory message = _fakeMessage();
+    address feeCollector = address(0x1);
+    bytes32 expectedEntryKey = inbox.sendL2Message{value: message.fee}(
+      message.recipient, message.deadline, message.content, message.secretHash
+    );
+
+    vm.expectEmit(true, false, false, false);
+    // event we expect
+    emit L1ToL2MessageCancelled(expectedEntryKey);
+    // event we will get
+    inbox.cancelL2Message(message, feeCollector);
+
+    // no such message to consume:
+    vm.expectRevert(
+      abi.encodeWithSelector(MessageBox.MessageBox__NothingToConsume.selector, expectedEntryKey)
+    );
+    inbox.get(expectedEntryKey);
+
+    // fees accrued as expected:
+    assertEq(inbox.feesAccrued(feeCollector), message.fee);
+  }
+
+  function testRevertIfNotConsumingFromRollup() public {
+    vm.prank(address(0x1));
+    bytes32[] memory entryKeys = new bytes32[](1);
+    entryKeys[0] = bytes32("random");
+    vm.expectRevert(MessageBox.MessageBox__Unauthorized.selector);
+    inbox.batchConsume(entryKeys, address(0x1));
+  }
+
+  function testRevertIfOneKeyIsPastDeadlineWhenBatchConsuming() public {
+    DataStructures.L1ToL2Msg memory message = _fakeMessage();
+    bytes32 entryKey1 = inbox.sendL2Message{value: message.fee}(
+      message.recipient, uint32(block.timestamp + 100), message.content, message.secretHash
+    );
+    bytes32 entryKey2 = inbox.sendL2Message{value: message.fee}(
+      message.recipient, uint32(block.timestamp + 200), message.content, message.secretHash
+    );
+    bytes32 entryKey3 = inbox.sendL2Message{value: message.fee}(
+      message.recipient, uint32(block.timestamp + 300), message.content, message.secretHash
+    );
+    bytes32[] memory entryKeys = new bytes32[](3);
+    entryKeys[0] = entryKey1;
+    entryKeys[1] = entryKey2;
+    entryKeys[2] = entryKey3;
+
+    skip(150); // block.timestamp now +150 ms. entryKey2 and entryKey3 is past deadline
+    vm.expectRevert(Inbox.Inbox__PastDeadline.selector);
+    inbox.batchConsume(entryKeys, address(0x1));
+  }
+
+  function testRevertIfConsumingAMessageThatDoesntExist(bytes32[] memory entryKeys) public {
```

Comment:
Would be useful to add an extra test trying to consume the same batch twice just to have it explicitly.